### PR TITLE
osd_types.cc: replace count with iterator in add_next_event function

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/432 Refs: #432

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
(replace tab with spaces)

File: src/osd/osd_types.h [hunk 1]
```diff
@@ -2371,6 +2371,7 @@ struct pg_missing_t {
   WRITE_CLASS_ENCODER(item)
 
   map<hobject_t, item> missing;         // oid -> (need v, have v)
+  map<hobject_t, item>::iterator missing_it;
```
Comment (liewegas) [inline src/osd/osd_types.h]: this should be local to the block where it is used in add_next_event()

**Comment by liewegas** [comment]
@dachary it's just a performance optimization... count() and then operator[] is duplicating the rbtree lookup.

**Comment by liewegas** [comment]
This is probably a good time to get rid of the commented-out assertions, too (in a separate patch).

**Comment by smithfarm** [comment]
@liewegas done, and done.

**Comment by liewegas** [comment]
Reviewed-by:

**Comment by tchaikov** [comment]
tested at
- http://pulpito.ceph.com/ubuntu-2015-06-29_18:58:24-rados-wip-kefu-testing---basic-multi/
- http://pulpito.ceph.com/ubuntu-2015-06-30_20:30:57-rados-wip-kefu-testing---basic-multi/

the failed ones are tracked at http://tracker.ceph.com/issues/12192, which are also spotted on the master branch.

**Pull request merged**